harvard-lts/drs-translation-service
# Retrieve a 'data ready' a message to the process queue

**Retrieve a 'data ready' a message to the process queue*
* * *

**GitHub Issue**: https://github.com/harvard-lts/HDC/issues/59)](https://github.com/harvard-lts/HDC/issues/70

# What does this Pull Request do?
Connects to the Process Queue and retrieves a data ready message from the queue

# How should this be tested?

1. Clone the repo
2. Checkout the branch HDC-70_process_queue_listening
3. Copy the env-template.txt to .env
4. Get the credentials from LPE Shared-HDC/Dev/QA ims ActiveMQ (transfer and process) for ims-process-dev
5. Add the credentials from step 4 to the .env for the PROCESS queueing (DRS is not needed for this)
6. Set .env variable `PROCESS_QUEUE_NAME=dvn-data-ready
7. Start up the docker container: `docker-compose -f docker-compose-local.yml up -d --build --force-recreate`
8. Exec into the container: `docker exec -it mqutils bash`
9. Run the command `pytest`
10. This will add a dummy message to the PROCESS_QUEUE and check that the listener retrieved it successfully.

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? No
- integration tests? Yes
